be/src/storage/lake/vertical_compaction_task.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "storage/lake/vertical_compaction_task.h"
+
+#include "runtime/runtime_state.h"
+#include "storage/chunk_helper.h"
+#include "storage/compaction_utils.h"
+#include "storage/lake/rowset.h"
+#include "storage/lake/tablet_metadata.h"
+#include "storage/lake/tablet_reader.h"
+#include "storage/lake/tablet_writer.h"
+#include "storage/lake/txn_log.h"
+#include "storage/row_source_mask.h"
+#include "storage/rowset/column_reader.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet_reader_params.h"
+#include "util/defer_op.h"
+
+namespace starrocks::lake {
+
+Status VerticalCompactionTask::execute(Progress* progress) {
```

Comment:
can this `VerticalCompactionTask::execute` interface called repeatedly?

Reply:
repeated calls are ok

Reply:
if calls repeatedly, those `_total_num_rows`, `_total_data_size` are still accurate?

Reply:
new task will be created when retry, so don't need to consider `execute` repeated calls.